controller/appcontroller.go

```diff
@@ -1675,6 +1671,13 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		return
 	}
 
+	destCluster, err := argo.GetDestinationCluster(context.Background(), app.Spec.Destination, ctrl.db)
+	if err != nil {
+		logCtx.Errorf("Failed to get destination cluster: %v", err)
+		// exit the reconciliation. ctrl.refreshAppConditions should have caught the error
+		return
+	}
+
```

Comment:
Should this be moved down just before `setAppManagedResources`, i.e. right before it's first used?

Reply:
I had it there, but If we fail getting the dest server, and it is somehow a transient error (because normal error should be caught by the validation above), then I think it is beneficial to skip all the compare app state logic and fail fast.